docs/protocol/extension_dynamic_logging.md

```diff
@@ -0,0 +1,177 @@
+<!--
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Dynamic Logging Extension
+
+This document describes Triton's dynamic logging extension. The logging extension enables
+the client to configure the log settings during a Triton run. Because this
+extension is supported, Triton reports TODO in the extensions field of
+its Server Metadata.
+
+## HTTP/REST
+
+In all JSON schemas shown in this document $number, $string, $boolean,
+$object and $array refer to the fundamental JSON types. #optional
+indicates an optional JSON field.
+
+Triton exposes the dynamic logging endpoint at the following URL. The client may use
+HTTP GET request to retrieve the current log settings. A HTTP POST request
+will modify the log settings, and the endpoint will return the updated log
+settings on success or an error in the case of failure. 
+
+```
+GET v2/dynamic_logging
+
+POST v2/dynamic_logging
+```
+
+### Log Setting Response JSON Object
+
+A successful log setting request is indicated by a 200 HTTP status
+code. The response object, identified as TODO, is
+returned in the HTTP body for every successful trace setting request.
+
+```
+$log_setting_response =
+{
+  $log_setting, ...
+}
+
+$log_setting = $string : $string | [ $string, ...]
```

Comment:
A verbal explanation of acceptable types might be helpful to users here. i.e.`$log_setting may be a string, or list of strings`, etc. (include booleans or other supported types as Guan suggests too if valid)

Reply:
Added remaining types and added those types to the description below.